### PR TITLE
Docstrings and minor fixes server

### LINKED ISSUE #3079: Outdated docstrings for cylc.network.server.ZMQServer
Going out soon and may not have time to prepare a pull request, but a low hanging fruit and happy if anyone beats me to it :-)

`cylc.network.server.py` has a few outdated docstrings (e.g. `start()` is still mentioning `ports` that was previously used, but now is using `min_port` and `max_port`).

Found while working on #3077 as the IDE raised a few new issues :+1: 

Cheers
Bruno

### PR DESCRIPTION
**Pull request by kinow**
Fix #3079 

- Adds docstrings fixing warnings that appeared in the IDE.
- Fixes a few typos
- Defines the instance `context` in the constructor

### CODE REVIEW

File: lib/cylc/network/server.py [hunk 1]
```diff
@@ -84,12 +85,13 @@ def __init__(self, encode_method, decode_method, secret_method):
...
-            ports (iterable): Generator of ports (int) to choose from.
-                The lowest available port will be chosen.
+        Will use a port range provided to select random ports.
 
+        Args:
+            min_port (int): minimum socket port number
+            max_port (int): maximum socket port number
         """
         # create socket
         self.context = zmq.Context()
```
Comment (oliver-sanders) [inline lib/cylc/network/server.py]: I think the context should probably be set in the `__init__` method. *(Looking at this code again the separation of `init` and `start` is purely for separation of concerns and not really functional, calling `start` twice would be nasty.)*

Comment (kinow) [reply lib/cylc/network/server.py]: :+1: good spot! Thanks!

File: lib/cylc/network/server.py [hunk 2]
```diff
@@ -84,15 +85,15 @@ def __init__(self, encode_method, decode_method, secret_method):
...
+        """Start the server running.
 
-        Args:
-            ports (iterable): Generator of ports (int) to choose from.
-                The lowest available port will be chosen.
+        Will use a port range provided to select random ports.
 
+        Args:
+            min_port (int): minimum socket port number
+            max_port (int): maximum socket port number
```
Comment (oliver-sanders) [inline lib/cylc/network/server.py]: BTW: we should make the site configuration more compatible with this, doing `min(ports), max(ports)` is wrong.

Comment (oliver-sanders) [reply lib/cylc/network/server.py]: Puted to #2978

**Pull request merged**